src/components/ActivityForm.js

```diff
@@ -27,57 +27,59 @@ export default class ActivityForm extends React.Component {
       name: props.activity ? props.activity.name : '',
       room: props.activity ? props.activity.room : '',
       teacher: props.activity ? props.activity.teacher : '',
-      error: '',
+      classNoError: '',
+      nameError: '',
+      roomError: '',
+      teacherError: '',
       // TODO: when you will use prop-types you will be able to improve this place ;)
       // TODO: I think, that now (looking at previous comments, maybe you will be able to
       // improve). If not, because activity will be able to be undefined, in separate PR
       // you should check lodash package and get method ;)
     };
   }
 
-  onNameChange = e => this.setState({ name: e.target.value });
-
-  onTeacherChange = e => this.setState({ teacher: e.target.value });
-
-  onRoomChange = e => this.setState({ room: e.target.value });
-
-  onClassNoChange = e => {
-    const classNo = e.target.value;
-    if (!classNo || classNo.match(/^[1-9]\d*$/)) {
-      this.setState({ classNo });
+  onInputValueChange = field => e => {
+    const inputValue = e.target.value;
+    if (field === 'classNo' && (!inputValue || inputValue.match(/^[1-9]\d*$/))) {
+      this.setState({ [field]: inputValue });
+    } else {
+      this.setState({ [field]: inputValue });
     }
+    this.setState({ [`${field}Error`]: '' });
   };
+  // this 'if' above looks weird but I think there's a way to specify the condition even more to avoid 'else' but I
```

Comment:
you can skip it. At first, you should merge setStates with error to avoid mutliple rerender. 
Second thing, it would be better to check if it is a classNo, and if it is not a number, just return from function.